src/idl_gen_general.cpp

```diff
@@ -409,6 +409,23 @@ static std::string GenDefaultValue(const LanguageParameters &lang, const Value &
            : value.constant;
 }
 
+static std::string GenEnumDefaultValue(const LanguageParameters &lang, const FieldDef &field) {
+  if (lang.language == GeneratorOptions::kCSharp) {
```

Comment:
you're already calling this from a context where the language is C#, no need to check double